mayastor/src/bdev/nexus/nexus_label.rs

```diff
@@ -126,340 +141,221 @@ pub enum LabelError {
-            ent_attr: 0,
-            ent_name: GptName {
-                name: "MayaMeta".into(),
-            },
-        };
-
-        entries[1] = GptEntry {
-            ent_type: GptGuid::from_str(Nexus::METADATA_PARTITION_TYPE_ID)
-                .unwrap(),
-            ent_guid: GptGuid::new_random(),
-            ent_start: entries[0].ent_end + 1,
-            ent_end: header.lba_end,
-            ent_attr: 0,
-            ent_name: GptName {
-                name: "MayaData".into(),
-            },
-        };
+        let partitions = Nexus::create_maya_partitions(
+            config,
+            &header,
+            block_size,
+            data_blocks,
+        )?;
 
-        header.table_crc = GptEntry::checksum(&entries);
+        header.table_crc = GptEntry::checksum(&partitions, header.num_entries);
         header.checksum();
 
-        //
         // Secondary GPT header
         let backup = header.to_backup();
 
-        NexusLabel {
+        Ok(NexusLabel {
             status: NexusLabelStatus::Neither,
             mbr: pmbr,
             primary: header,
-            partitions: entries,
+            partitions,
             secondary: backup,
-        }
-    }
-
-    fn get_primary_data(
-        &self,
-        label: &NexusLabel,
-    ) -> Result<LabelData, LabelError> {
-        let block_size = self.bdev.block_len() as u64;
-        let blocks = Aligned::get_blocks(
-            GPTHeader::PARTITION_TABLE_SIZE as u64,
-            block_size,
-        );
-        // allocate 2 extra blocks for the MBR and GPT header respectively
-        let mut buf =
-            DmaBuf::new((blocks + 2) * block_size, self.bdev.alignment())
-                .context(WriteAlloc {
-                    name: String::from("primary"),
-                })?;
-        let mut writer = Cursor::new(buf.as_mut_slice());
-
-        // Protective MBR
-        writer.seek(SeekFrom::Start(440)).unwrap();
-        serialize_into(&mut writer, &label.mbr).context(SerializeError {})?;
-
-        // Primary GPT header
-        writer
-            .seek(SeekFrom::Start(label.primary.lba_self * block_size))
-            .unwrap();
-        serialize_into(&mut writer, &label.primary)
-            .context(SerializeError {})?;
-
-        // Primary partition table
-        writer
-            .seek(SeekFrom::Start(label.primary.lba_table * block_size))
-            .unwrap();
-        for entry in &label.partitions {
-            serialize_into(&mut writer, &entry).context(SerializeError {})?;
-        }
-
-        Ok(LabelData {
-            offset: 0,
-            buf,
         })
     }
 
-    fn get_secondary_data(
-        &self,
-        label: &NexusLabel,
-    ) -> Result<LabelData, LabelError> {
-        let block_size = self.bdev.block_len() as u64;
-        let blocks = Aligned::get_blocks(
-            GPTHeader::PARTITION_TABLE_SIZE as u64,
-            block_size,
+    // Create partition table entries for the MayaMeta and
```

Comment:
This is probably better served as a `///` comment for IDE tooling.

Reply:
Done